// ngx-extended-pdf-viewer/esm2015/lib/sidebar/pdf-sidebar/pdf-sidebar.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/sidebar/pdf-sidebar/pdf-sidebar.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, TemplateRef, Output, EventEmitter } from '@angular/core';
export class PdfSidebarComponent {
    constructor() {
        this.showSidebarButton = true;
        this.thumbnailDrawn = new EventEmitter();
    }
}
PdfSidebarComponent.decorators = [
    { type: Component, args: [{
                selector: 'pdf-sidebar',
                template: "<div>\n  <ng-content *ngTemplateOutlet=\"customSidebar ? customSidebar : defaultSidebar\"></ng-content>\n</div>\n\n<ng-template #defaultSidebar>\n  <div id=\"sidebarContainer\" [style.top]=\"sidebarPositionTop\">\n    <div id=\"additionalSidebarContainer\" [class.invisible]=\"!showSidebarButton\">\n      <pdf-sidebar-toolbar></pdf-sidebar-toolbar>\n    </div>\n    <pdf-sidebar-content [customThumbnail]=\"customThumbnail\" (thumbnailDrawn)=\"thumbnailDrawn.emit($event)\"></pdf-sidebar-content>\n    <div id=\"sidebarResizer\" class=\"hidden\"></div>\n  </div>\n</ng-template>\n\n",
                styles: [""]
            }] }
];
/** @nocollapse */
PdfSidebarComponent.ctorParameters = () => [];
PdfSidebarComponent.propDecorators = {
    sidebarPositionTop: [{ type: Input }],
    showSidebarButton: [{ type: Input }],
    customSidebar: [{ type: Input }],
    customThumbnail: [{ type: Input }],
    thumbnailDrawn: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    PdfSidebarComponent.prototype.sidebarPositionTop;
    /** @type {?} */
    PdfSidebarComponent.prototype.showSidebarButton;
    /** @type {?} */
    PdfSidebarComponent.prototype.customSidebar;
    /** @type {?} */
    PdfSidebarComponent.prototype.customThumbnail;
    /** @type {?} */
    PdfSidebarComponent.prototype.thumbnailDrawn;
}
